test/OpenTelemetry.Api.Tests/Trace/Propagation/BaggagePropagatorTest.cs

```diff
@@ -118,50 +117,100 @@ public void ValidateLongBaggageExtraction()
+        };
+
+        var propagationContext = this.baggage.Extract(default, carrier, GetterList);
+
+        Assert.False(propagationContext == default);
+        Assert.True(propagationContext.ActivityContext == default);
 
-        Assert.Equal("key2", encodedKey);
-        Assert.Equal("!x_x%2Cx-x%26x(x%22)%3B%3A", encodedValue);
-        Assert.Equal("key%28%293", escapedKey);
-        Assert.Equal("value%28%29%21%26%3B%3A", escapedValue);
+        Assert.Equal(1, propagationContext.Baggage.Count);
 
-        var initialBaggage =
-            $"key%201=value%201,{encodedKey}={encodedValue},{escapedKey}={escapedValue},key4=%20%21%22%23%24%25%26%27%28%29%2A%2B%2C-.%2F0123456789%3A%3B%3C%3D%3E%3F%40ABCDEFGHIJKLMNOPQRSTUVWXYZ%5B%5C%5D%5E_%60abcdefghijklmnopqrstuvwxyz%7B%7C%7D~";
+        var actualBaggage = propagationContext.Baggage.GetBaggage();
+
+        Assert.Single(actualBaggage);
 
+        Assert.Contains(key, actualBaggage);
+        Assert.Equal(expectedDecodedValue, actualBaggage[key]);
+    }
+
+    [Fact]
+    public void ValidateBaggageMultipleItemsExtraction()
+    {
         var carrier = new List<KeyValuePair<string, string>>
         {
-            new KeyValuePair<string, string>(BaggagePropagator.BaggageHeaderName, initialBaggage),
+            new(BaggagePropagator.BaggageHeaderName, "  key1  =val1,key2=  val2  ,key3="),
         };
 
         var propagationContext = this.baggage.Extract(default, carrier, GetterList);
 
         Assert.False(propagationContext == default);
         Assert.True(propagationContext.ActivityContext == default);
 
-        Assert.Equal(4, propagationContext.Baggage.Count);
-
         var actualBaggage = propagationContext.Baggage.GetBaggage();
 
-        Assert.Equal(4, actualBaggage.Count);
+        Assert.Equal(3, actualBaggage.Count);
 
-        Assert.True(actualBaggage.ContainsKey("key 1"));
-        Assert.Equal("value 1", actualBaggage["key 1"]);
+        Assert.True(actualBaggage.ContainsKey("key1"));
+        Assert.Equal("val1", actualBaggage["key1"]);
 
         Assert.True(actualBaggage.ContainsKey("key2"));
-        Assert.Equal("!x_x,x-x&x(x\");:", actualBaggage["key2"]);
+        Assert.Equal("val2", actualBaggage["key2"]);
 
-        Assert.True(actualBaggage.ContainsKey("key()3"));
-        Assert.Equal("value()!&;:", actualBaggage["key()3"]);
+        Assert.True(actualBaggage.ContainsKey("key3"));
+        Assert.Equal(string.Empty, actualBaggage["key3"]);
+    }
 
-        // x20-x7E range
-        Assert.True(actualBaggage.ContainsKey("key4"));
-        Assert.Equal(" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~", actualBaggage["key4"]);
+    [Theory]
+    [InlineData("key(=value")]
+    [InlineData(@"key=\\")]
+    [InlineData("key=val%")]
+    [InlineData("key=%gg")]
+    [InlineData("key=val%2")]
+    [InlineData("key=val ue")]
+    [InlineData("key=val%IJK")]
```

Comment:
Add a case which has valid percent coding triplets, but invalid as value in list-member once URI escaped?